### PR TITLE
Collision: Code cleanup

### PR DESCRIPTION
**Pull request by SmallJoker**
Proper floating point calculations in `collisionMoveSimple`
Add helper function `sortBoxVerticies` for `vector3df`
Minor code style cleanups

2nd commit adds another special group to modify the damage tolerance of falling. Enjoy.
Warning: Older clients are not aware of this functionality!

### CODE REVIEW

File: src/clientenvironment.cpp [hunk 1]
```diff
@@ -215,32 +215,29 @@ void ClientEnvironment::step(float dtime)
...
 		f32 post_factor = 1; // 1 hp per node/s
-		if(info.type == COLLISION_NODE)
-		{
+		if (info.type == COLLISION_NODE) {
 			const ContentFeatures &f = m_client->ndef()->
 				get(m_map->getNodeNoEx(info.node_p));
 			// Determine fall damage multiplier
 			int addp = itemgroup_get(f.groups, "fall_damage_add_percent");
-			pre_factor = 1.0 + (float)addp/100.0;
+			pre_factor += (float)addp / 100.f;
```
Comment (Zeno-) [inline src/clientenvironment.cpp]: Since you're cleaning up, casting addp to float is not necessary and just (IMO) makes the line harder to read

**Comment by paramat** [comment]
Tolerance in nodes fallen is difficult to work with in code, needing calculation involving custom gravity, how about a tolerated impact speed instead?

**Comment by SmallJoker** [comment]
The new, effective gravity is already contain in the variable. However, I could change this behaviour, so it bases on the velocity instead of fixed nodes. Players would have to calculate the height themselves if needed.

**Comment by paramat** [comment]
Nodes of height is more player-friendly, i was just concerned about whether this made the core code complex, presumably all that is known on impact is speed anf gravity, how to convert these into nodes fallen?

**Comment by paramat** [comment]
```
+				// Gravity calculation is somewhat broken
+				// Average fall speed is 1/4 the final speed (wtf?)
```
This is not necessarily a error, the player is accelerating and it depends on how the 'average' is calculated.
Although i'm not sure yet i'm suspicious that this code is correct, calculating nodes fallen from impact speed and gravity takes quite a lot of complex physics calculation that i'm not sure about despite being a physicist. There may be other factors that accelerate a player that makes this inaccurate. It would be simpler and more accurate to just specify a tolerated impact velocity.
For the moment then :-1: for how tolerance works.

**Comment by paramat** [comment]
Should this replace 'fall damage add percent' feature? or is this needed if we already have that?

**Comment by SmallJoker** [comment]
Reverted the feature to speed up the process.

File: src/collision.cpp [hunk 2]
```diff
@@ -317,13 +315,12 @@ collisionMoveResult collisionMoveSimple(Environment *env, IGameDef *gamedef,
 			}
 			std::vector<aabb3f> nodeboxes;
 			n.getCollisionBoxes(gamedef->ndef(), &nodeboxes, neighbors);
-			for(std::vector<aabb3f>::iterator
+			for (std::vector<aabb3f>::iterator
```
Comment (nerzhul) [inline src/collision.cpp]: can you use a C++11 for loop here ? for (aabb3f &box: nodeboxes)

**Comment by SmallJoker** [comment]
@nerzhul There you go. Rebased the whole thing aswell.

**Pull request closed without merging**